fastlane/lanes/build.rb

```diff
@@ -85,13 +85,14 @@
       path.include?(options[:name])
     end.first
 
-    UI.user_error!("Unable to find .xctestrun file at #{build_products_path}") if xctestrun_path.nil? || !File.exist?((xctestrun_path))
+    UI.user_error!("Unable to find .xctestrun file at #{build_products_path}.") if xctestrun_path.nil? || !File.exist?((xctestrun_path))
 
     inject_buildkite_analytics_environment(xctestrun_path: xctestrun_path) if buildkite_ci?
+    scheme_name = (!options[:name] == 'Jetpack' ? 'JetpackUITests' : 'WordPress')
```

Comment:
This condition seems a bit hard to read… and thus I'm not sure it's correct.

Does that mean "if it's not true that options[:name] == 'Jetpack'"? But if so, that means that if `options[:name]` is _not_ `'Jetpack'`, then you use `JetpackUITests`… shouldn't that be the opposite?

Reply:
@jostnes the `name` parameter is "The (partial) name of the `*.xctestrun` file to run".

Those `.xctestrun` files are generated by `build_<app>_for_testing` and contain the name of the `xctestfile` files from the project.

E.g. On my machine `build_jetpack_for_testing` generates:

```
➜ ls DerivedData/Build/Products/*.xctestrun
DerivedData/Build/Products/Jetpack_JetpackUITests_iphonesimulator16.2-arm64.xctestrun
```

Therefore, I think we could simplify this logic for the time to something like

```suggestion
    # Our current configuration allows for either running the Jetpack UI tests or the WordPress unit tests.
    #
    # Their scheme and xctestrun name are:
    #
    # (JetpackUITests, JetpackUITests)
    # (WordPress, WordPressUnitTests)
    #
    # Because we only support those two modes, we can infer the scheme name from the xctestrun name
    scheme_name = options[:name].include? 'Jetpack' ? 'JetpackUITests' : 'WordPress'
```

Reply:
Also, but this is a nitpick, `scheme_name` could be shortened to `scheme` 🤓

Reply:
@AliSoftware looking at it again, it does look incorrect (though CI passed 🤔) and i agree that it's hard to read, it went through a few iterations of changes and ended up being hard to read - should be updated now (see below for commits)

and thank you for the suggestion @mokagio! i think that reads better, updated in https://github.com/wordpress-mobile/WordPress-iOS/pull/20301/commits/321a456c4212a4ed69de1d757754baf7509e8317 (update: there was lint error so i updated it again to use boolean expression in https://github.com/wordpress-mobile/WordPress-iOS/pull/20301/commits/26fadd08f99c52424d77fb04918279b4c3b1f794)